patches/@vscode__extension-telemetry@0.6.2.patch

```diff
@@ -0,0 +1,13 @@
+ `+o.message)}},logException:(a,s)=>{throw new Error("1DS SDK does not support logging exceptions, please use logEvent for exception tracking")},flush:async()=>{try{n==null||n.unload()}catch(a){throw new Error(`Failed to flush app insights!
+-`+a.message)}}}};var Gx=async(e,t)=>{let r;try{process.env.APPLICATION_INSIGHTS_NO_DIAGNOSTIC_CHANNEL="1";let i=await Promise.resolve().then(()=>nn(nA()));i.defaultClient?(r=new i.TelemetryClient(e),r.channel.setUseDiskRetryCaching(!0)):(i.setup(e).setAutoCollectRequests(!1).setAutoCollectPerformance(!1).setAutoCollectExceptions(!1).setAutoCollectDependencies(!1).setAutoDependencyCorrelation(!1).setAutoCollectConsole(!1).setAutoCollectHeartbeat(!1).setUseDiskRetryCaching(!0).start(),r=i.defaultClient),Gt&&Gt.env&&(r.context.tags[r.context.keys.userId]=Gt.env.machineId,r.context.tags[r.context.keys.sessionId]=Gt.env.sessionId,r.context.tags[r.context.keys.cloudRole]=Gt.env.appName,r.context.tags[r.context.keys.cloudRoleInstance]=Gt.env.appName),e&&e.indexOf("AIF-")===0&&(r.config.endpointUrl="https://mobile.events.data.microsoft.com/collect/v1")}catch(i){return Promise.reject(`Failed to initialize app insights!
++`+a.message)}}}};var Gx=async(e,t)=>{let r;try{process.env.APPLICATION_INSIGHTS_NO_DIAGNOSTIC_CHANNEL="1";let i=await Promise.resolve().then(()=>nn(nA()));i.defaultClient?(r=new i.TelemetryClient(e),r.channel.setUseDiskRetryCaching(!0)):(i.setup(e).setAutoCollectRequests(!1).setAutoCollectPerformance(!1).setAutoCollectExceptions(!1).setAutoCollectDependencies(!1).setAutoDependencyCorrelation(!1).setAutoCollectConsole(!1).setAutoCollectHeartbeat(!1).setUseDiskRetryCaching(!0).start(),r=i.defaultClient),Gt&&Gt.env&&(r.context.tags[r.context.keys.userId]=Gt.env.machineId,r.context.tags[r.context.keys.sessionId]=Gt.env.sessionId,r.context.tags[r.context.keys.cloudRole]=Gt.env.appName,r.context.tags[r.context.keys.cloudRoleInstance]=Gt.env.appName,r.context.tags["ai.location.ip"]="0.0.0.0"),e&&e.indexOf("AIF-")===0&&(r.config.endpointUrl="https://mobile.events.data.microsoft.com/collect/v1")}catch(i){return Promise.reject(`Failed to initialize app insights!
```

Comment:
For the records, this patch was required to clear the IP address so no location information is logged. We've added:
```javascript
r.context.tags["ai.location.ip"]="0.0.0.0"
```

I've cloned https://github.com/microsoft/vscode-extension-telemetry and checked the code, but did not find a way to overwrite `ai.location.ip`. Issue https://github.com/microsoft/vscode-extension-telemetry/issues/128 talks about this, but the proposed solution using `replacementOptions` does not work, as `ai.location.ip` is not exposed as property key. Perhaps it is added later on a lower level?

We can see  and replace these properties only:

<img width="492" alt="image" src="https://user-images.githubusercontent.com/66327622/216213454-82911e00-3641-4e82-b010-2ddaf3d6c5fe.png">

https://github.com/microsoft/vscode-extension-telemetry/blob/04ff08da26bc437e87bfb398ac63be2685576066/src/common/util.ts#L45-L57

The replacement options do not allow to set a value that does not exist yet and the `appInsightsClient`, on which it could be also set manually can not be accessed through `TelemetryReporter`.